src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -680,15 +681,15 @@ private function resolveServices($value, $file, $isParameter = false)
     {
         if ($value instanceof TaggedValue) {
             $argument = $value->getValue();
-            if ('iterator' === $value->getTag()) {
+            if ('service_locator' === $value->getTag() || 'iterator' === $value->getTag()) {
                 if (!is_array($argument)) {
-                    throw new InvalidArgumentException(sprintf('"!iterator" tag only accepts sequences in "%s".', $file));
+                    throw new InvalidArgumentException(sprintf('"!%s" tag only accepts sequences in "%s".', $value->getTag(), $file));
```

Comment:
In xml each service is keyed, can't they be in yaml? If yes, `service_locator` also accepts mappings (maybe only mappings, not sequences?). c.f. https://github.com/symfony/symfony/pull/21553#pullrequestreview-20616879 (glad to see the argument back btw :))

Reply:
indeed, a service locator accepts maps, not sequences.